checks/raw/security_policy.go

```diff
@@ -44,7 +45,15 @@ func SecurityPolicy(c *checker.CheckRequest) (checker.SecurityPolicyData, error)
 		return checker.SecurityPolicyData{}, err
 	}
 	// If we found files in the repo, return immediately.
+	// TODO: // assert that at this point only 1 file is returned by isSecurityPolicyFile
 	if len(data.files) > 0 {
+		err := fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
```

Comment:
I think you can update the existing code to perform the matching as well, instead of doing that at the end.
What I mean is that you can update previous calls to `OnAllFilesDo` to `OnMatchingFileContentDo`

Reply:
This is one that I am not sure of.

What I found with `OnMatchingFileContentDo` (OMFCD), I would set the (regex) pattern to be the name of **the** file I want, e.g., `security.md` (like that returned from `isSecurityPolicyFilename`) having to "pattern" (per se), and OMFCD would return that **and** any other file within the repo that had that name.... I recall seeing `/<dir>/<dir>/<dir>/<dir>/security.md` which were not wanted.

This is because OMFCD uses `isMatchingPath` walking the entire repo, and finds **all** files which (in this case) have `security.md`--because `isMatchingPath` uses two tests (see lines 36-47 of `checks/fileparser/listing.go`) one on the fullpath and the other on the filename (or basename) for the search. `OnAllFilesDo` (OAFD) does not do that (use `isMatchingPath`)

So, for the security policy, say I was searching for `security.adoc` and not using (OAFD) **and** `security.adoc` was not in `repos root`, `.github/`, or `docs/` but did appear deep in the repo, say `/contrib/test/other/data/whatif/security.adoc`, OMFCD would actually provide the contents of that file which would then be parsed/evaluated.

What seems to work is keying off `OnAllFilesDo` finding what I am looking for, using the specific pattern to OMFCD and stopping at the first file match.

Perhaps I am using OMFCD incorrectly, and need hint as to a better way to use OMFCD.

Reply:
wait... actually, (I wonder) if OMFCD could be parameterized (perhaps a `bool` like `CaseSensitive`) to NOT do the backup `filename` and stick to the first search for full path, then I think OMFCD could / should work. That would mean that `isSecurityPolicyFilename` behavior would change (unless OMFCD could also be parameterized with a predicate like `isSecurityPolicyFilename`) -- just a thought

Reply:
I have not tried to optimize this (yet) - it is an optimization which can be performed.

One potential thought I had was rather than use OMFCD, keep using OAFD with `repoClient.GetFileContent` (which is used within OMFCD) and just get the contents directly.